src/main/java/nextstep/autoconfigure/FilterOrderRegistration.java

```diff
@@ -0,0 +1,57 @@
+package nextstep.autoconfigure;
+
+import jakarta.servlet.Filter;
+import nextstep.oauth2.web.OAuth2AuthorizationRequestRedirectFilter;
+import nextstep.oauth2.web.OAuth2LoginAuthenticationFilter;
+import nextstep.security.authentication.BasicAuthenticationFilter;
+import nextstep.security.authentication.UsernamePasswordAuthenticationFilter;
+import nextstep.security.authorization.AuthorizationFilter;
+import nextstep.security.context.SecurityContextHolderFilter;
+import nextstep.security.csrf.CsrfFilter;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FilterOrderRegistration {
+    private final Map<Class<? extends Filter>, Integer> filterToOrder = new HashMap<>();
+
+    public FilterOrderRegistration() {
+        Step order = new Step(100, 100);
+        filterToOrder.put(CsrfFilter.class, order.next());
+        filterToOrder.put(SecurityContextHolderFilter.class, order.next());
+        filterToOrder.put(UsernamePasswordAuthenticationFilter.class, order.next());
+        filterToOrder.put(BasicAuthenticationFilter.class, order.next());
+        filterToOrder.put(OAuth2AuthorizationRequestRedirectFilter.class, order.next());
+        filterToOrder.put(OAuth2LoginAuthenticationFilter.class, order.next());
+        filterToOrder.put(AuthorizationFilter.class, order.next());
+    }
+
+    public Integer getOrder(Class<? extends Filter> clazz) {
+        Integer order = filterToOrder.get(clazz);
+        if (order == null) {
+            return Integer.MAX_VALUE;
+        }
+        return order;
+    }
+
+    public void put(final Filter filter, final int order) {
+        filterToOrder.put(filter.getClass(), order);
```

Comment:
putIfAbsent()를 활용하여, 이미 등록된 필터의 순서를 변경하지 않도록 수정하면 더 안전한 구조로 개선해볼 수 있겠네요 😄 
```suggestion
        filterToOrder.putIfAbsent(filter.getClass(), order);
```

Reply:
이런 부분이 디테일인데 앞으로도 신경써야겠습니다